crates/ide/src/syntax_highlighting/highlight.rs

```diff
@@ -108,6 +109,11 @@ pub(super) fn element(
                         highlight_name_ref_by_syntax(name_ref, sema)
                     }
                     None => HlTag::UnresolvedReference.into(),
+                };
+                if h.tag == HlTag::Symbol(SymbolKind::Module) && is_self {
+                    HlTag::Symbol(SymbolKind::SelfParam).into()
+                } else {
+                    h
```

Comment:
Unfortunately a special case still remains for `extern crate self` and `self` in paths so they don't get lit as modules.